implementation/README.md

```diff
@@ -0,0 +1,147 @@
+
+#### Working with a system that only has a net interface
+
+If the system only has a net interface, you use Dynamic Host Configuration Protocol (DHCP), using broadcast DISCOVER, and Trivial File Transfer Protocol (TFTP) to get the boot information you need. These programs are provided by the PXEboot program.
+
+Configuration information is provided by REPLY to a DHCP request. The REPLY returns an IP, server, and a configuration file name that provides:
+
+   +   Identity
+   +   What to boot
+   +   Where to get it
+
+Data is provided by TFTP. On proper systems, HTTP downloading takes a fraction of a second even for 16M kernels. With TFTP it's very slow and TFTP won't work with initramfs much large than 32MiB. Most LinuxBoot shops use or are transitioning to HTTP.
+
+Note: Boot images require a kernel(bzImage) + an initramfs + a command line. They can be loaded as three pieces or compiled and loaded as one piece, as described in this section.
+
+### Step 2: replace Shell binary section
+
++   UEFI Shell is a DXE
+
+    +   DXEs are Portable Executable 32-bit binaries (PE32)
+    +   They have multiple sections, one of them being binary code
+    +   You need a flash image (in this case called firmware.bin). You can get it via vendor website, flashrom, or other mechanism.
+
++   The following `utk` command replaces the Shell code section with a Linux kernel:
+
+      ` utk firmware.bin replace_pe32 Shell bzImage save new.bin`
+
+     Note: it's always a PE32, even for 64-bit kernels
+
++   After running `utk`, you can reflash
+
+### Step 3: remove as many DXEs as possible
+
++   You can do an initial mass removal based on your current knowledge
++   `utk` automates removing DXEs: this is the DXE cleaner
+
+    +   `utk` removes a DXE, reflashes, checks if it boots, repeats
+
+    This part should be easy: DXE can have a dependency section. In practice, it's hard: because dependency sections are full of errors and omissions. Sloppy UEFI code frequently does not check for failed DXE loads, e.g., remove EnglishDXE, get a General Protection Fault.
+
+### Step 4: replace closed-source with open source
+
++   If you can build a DXE from source, you can use `utk` to remove the proprietary one and replace it with one built from source. You can get DXE source from the tianocore/EDK2 source repo at github.com. The GitHub repo has a **_limited_** number of DXEs in source form; i.e., you can't build a full working image using it.
++   There are scripts that let you compile individual DXEs, including the UEFI Shell and Boot Device Selection (BDS). These two DXEs have been compiled and are used in the Atomic Pi. Source-based BDS was needed to ensure the UEFI Shell was called.
++   You only need the UEFI Shell built long enough to replace it with Linux.
+
+### Final step: reflash the image
+
++   "Native" reflash: Boot the system whatever way is easiest: netboot, usb, local disk, and run:
+
+    `flashrom -p internal -w new.bin`
```

Comment:
where new.bin is a standin a file name which might be anything, what's the right way to say that.

Reply:
I am addressing the issue in the updated PR. Let me know if it does not work as you'd like.